src/components/services-view/service-view.classes.ts

```diff
@@ -0,0 +1,193 @@
+import * as angular from 'angular';
+import {allCategories} from '../../constants';
+import * as _ from 'lodash';
+
+interface IServiceItem {
+    iconClass: string;
+    imageUrl: string;
+    name: string;
+    catsBySubCats: any;
+    description: string;
+    longDescription: string;
+    data: any;
+}
+
+export class ServiceItem implements IServiceItem {
+    public iconClass: string;
+    public imageUrl: string;
+    public name: string;
+    public catsBySubCats: any;
+    public description: string;
+    public longDescription: string;
+    public data: any;
+
+    constructor (serviceClass: any) {
+        this.data = serviceClass;
+        this.imageUrl = this.getImage();
+        this.name = this.getName();
+        this.description = this.getDescription();
+        this.longDescription = this.getLongDescription();
+        this.catsBySubCats = this.getCategoriesBySubCategories();
+    }
+
+    private getImage() {
+        if (this.data.metadata) {
+            return this.data.metadata.imageUrl;
+        }
+    }
+
+    private getName() {
+        if (this.data.osbMetadata) {
+            return this.data.osbMetadata.displayName ? this.data.osbMetadata.displayName : this.data.metadata.name;
+        } else {
+            return this.data.metadata.name;
+        }
+    }
+
+    private getDescription() {
+        if (this.data.osbMetadata) {
+            return this.data.osbMetadata.description;
+        }
+    }
+
+    private getLongDescription() {
+        if (this.data.osbMetadata) {
+            return this.data.osbMetadata.longDescription;
+        }
+    }
+
+    private getCategoriesBySubCategories() {
+        let catsBySubCats = {};
+
+        if (this.data.osbTags) {
+            _.each(this.data.osbTags, (subCategory) => {
+                let subCat = getSubCategoryFromAlias(subCategory);
+                let cat = getCategoryFromSubCategory(subCat);
+                catsBySubCats[subCat] = cat;
+            });
+        }
+        return catsBySubCats;
+    }
+}
+
+export class ImageItem implements IServiceItem {
+    public iconClass: string;
+    public imageUrl: string;
+    public name: string;
+    public catsBySubCats: any;
+    public description: string;
+    public longDescription: string;
+    public data: any;
+    public isBuilder: boolean;
+    private annotations: any;
+    private $filter: any;
+
+    constructor (image: any, $filter: any) {
+        this.data = image;
+        this.annotations = this.getAnnotations();
+        this.iconClass = this.getIcon();
+        this.name = this.getName();
+        this.description = this.getDescription();
+        this.longDescription = this.getLongDescription();
+        this.catsBySubCats = this.getCategoriesBySubCategories();
+        this.$filter = $filter;
+        console.log("construct $filter = " + JSON.stringify($filter, null, ' '));
```

Comment:
Arrgg...my apologies!  I meant to remove 'service-view.classes.ts'!  It has been replaced with 'src/services/catalog.service.ts'